File: src/main/scala/com/codebetyars/skyhussars/engine/MainMenu.scala
package com.codebetyars.skyhussars.engine

import de.lessvoid.nifty.Nifty
import de.lessvoid.nifty.screen.{Screen, ScreenController}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import scala.beans.BeanProperty
//remove if not needed

@Component
class MainMenu extends GameState with ScreenController {

  @Autowired
  private var guiManager: GuiManager = _

  @BeanProperty
  var pendingMission: GameState = _

  private var time: Float = 0

  override def update(tpf: Float): GameState = {
    var nextState: GameState = this
    if (guiManager.gameStarted) {
      nextState = pendingMission
      guiManager.startGame()
    }
    nextState
  }

  override def close() {
  }

  override def initialize() {
    guiManager.stopGame()
    guiManager.cursor(true)
  }

  def startGame() {
    guiManager.startGame()
  }

  def bind(nifty: Nifty, screen: Screen) {
  }

  def onStartScreen() {
  }

  def onEndScreen() {
  }
}

File: src/main/java/com/codebetyars/skyhussars/engine/MainMenu.java
/*
 * Copyright (c) 2016, ZoltanTheHun
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * * Redistributions of source code must retain the above copyright notice, this
 *   list of conditions and the following disclaimer.
 * * Redistributions in binary form must reproduce the above copyright notice,
 *   this list of conditions and the following disclaimer in the documentation
 *   and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
 * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
 * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
 * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
 * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
 * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
 * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGE.
 */

package com.codebetyars.skyhussars.engine;

import de.lessvoid.nifty.Nifty;
import de.lessvoid.nifty.screen.Screen;
import de.lessvoid.nifty.screen.ScreenController;

public class MainMenu extends GameState implements ScreenController {

    public MainMenu(GuiManager guiManager, GameState state) {
        this.state = state;
        this.guiManager = guiManager;
    }
    private GameState state;
    private GuiManager guiManager;
    private float time = 0;
    //private boolean startGame = false;

    @Override
    public GameState update(float tpf) {
        GameState nextState = this;
        if (guiManager.startGame) {
            nextState = state;
            guiManager.startGame = false;
        }
        return nextState;
    }

    @Override
    public void close() {
    }

    @Override
    public void initialize() {
        guiManager.startGame = false;
        guiManager.cursor(true);
    }

    public void startGame() {
        guiManager.startGame = true;
    }

    public void bind(Nifty nifty, Screen screen) {
    }

    public void onStartScreen() {
    }

    public void onEndScreen() {
    }
}
